util/src/main/java/io/kubernetes/client/util/WebSocketStreamHandler.java

```diff
@@ -265,10 +272,25 @@ public void write(byte[] b, int offset, int length) throws IOException {
         int bufferSize = Math.min(remaining, 15 * 1024 * 1024);
         byte[] buffer = new byte[bufferSize + 1];
         buffer[0] = stream;
+
         System.arraycopy(b, offset + bytesWritten, buffer, 1, bufferSize);
-        if (!WebSocketStreamHandler.this.socket.send(ByteString.of(buffer))) {
+        ByteString byteString = ByteString.of(buffer);
+
+        int attempts = 0;
+        while (WebSocketStreamHandler.this.socket.queueSize() + byteString.size() > MAX_QUEUE_SIZE
+            && attempts < MAX_QUEUE_SIZE_MAX_ATTEMPTS) {
+          try {
+            Thread.sleep(MAX_QUEUE_SIZE_WAIT_MILLISECONDS);
+            attempts++;
```

Comment:
Instead of tracking max number of attempts, let's track `MAX_WAIT_MILLIS`

e.g. make the condition: `while( now() < start + MAX_WAIT_MILLIS) { ... }`

I think that is easier to reason about and cleaner code too. (also when you combine it with `wait/notify` it's even cleaner.

Reply:
Done!